src/routes/genre.routes.ts

```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import * as genreControllers from '../controllers/genreControllers';
+import * as genreValidationRules from '../middleware/genreValidationRules';
+
+const router = Router();
+
+router.get('/', genreControllers.getAllGenres);
+router.post('/', genreControllers.createGenre, genreValidationRules.genreCreateRules);
+router.patch('/:id', genreControllers.updateGenre, genreValidationRules.genreUpdateRules);
+router.delete('/:id', genreControllers.deleteGenre, genreValidationRules.genreDeleteRules);
```

Comment:
Are you sure the validation middleware should be called after the controller is called? Validation will not be called in most cases, as the controller will return the result before passing control to the next middleware. The primary purpose of validation is to validate input data before processing. Please check this point through your code.

Reply:
Yes, I've changed it